plugins/out_loki/loki.c

```diff
@@ -320,7 +338,17 @@ static flb_sds_t pack_labels(struct flb_loki *ctx, msgpack_packer *mp_pck,
                 /* append the key/value pair */
                 flb_mp_map_header_append(&mh);
                 msgpack_pack_str(mp_pck, k.via.str.size);
+
+#if FLB_LOKI_ESCAPE_KUBERNETES_AUTO_KEYS == 1
+                char *dkey = strndup(k.via.str.ptr, k.via.str.size);
```

Comment:
I think this need a different approach, since it always assumes that a special character exists so a memory copy is needed, this actually needs to be avoided.

Reply:
Sadly the in place replacement is not possible since it modifies the data too (referenced key processing). Or do you think pre-checking the key for special chars first would be more efficient?